app/game/game.js

```diff
@@ -111,10 +118,20 @@ const GAME = (humanUsername, aiUsername, gameId) => {
     }
 
     const onThreatUnresolved = (room) => {
-        delete THREATS[THREATS_INDEXED_BY_ROOM] // Remove threat from THREATS object
-        AVAILABLE_ROOMS.splice(AVAILABLE_ROOMS.indexOf(room), 1);
+        delete THREATS_INDEXED_BY_ROOM[room] // Remove threat
+        AVAILABLE_ROOMS.splice(AVAILABLE_ROOMS.indexOf(room), 1); // Room no longer available
+        ROOMS_WITH_THREATS.splice(ROOMS_WITH_THREATS.indexOf(room), 1); // Removes room from rooms_with_threats
         console.log(`Threat was unresolved room ${room} is no longer available`);
     }
+    const onThreatResolved = (room) => {
+        delete THREATS_INDEXED_BY_ROOM[room] // Remove threat
+        ROOMS_WITH_THREATS.splice(ROOMS_WITH_THREATS.indexOf(room), 1); // Removes room from rooms_with_threats
```

Comment:
These lines are duplicated above. Let's split the shared code from `onThreatUnresolved()` and `onThreatResolved()` out into a `removeThreat()`